http/codegen/service_data.go

```diff
@@ -2184,7 +2191,7 @@ func extractPathParams(a *expr.MappedAttributeExpr, service *expr.AttributeExpr,
 			StringSlice:    arr != nil && arr.ElemType.Type.Kind() == expr.StringKind,
 			Map:            false,
 			MapStringSlice: false,
-			Validate:       codegen.RecursiveValidationCode(c, ctx, true, varn),
+			Validate:       codegen.RecursiveValidationCode(c, ctx, true, varn, name),
```

Comment:
Setting original path param name, to avoid setting escaped name to the validation target.

```
err = goa.MergeErrors(err, goa.ValidatePattern("c2", c2, "patternc")) 
```
after:
```
err = goa.MergeErrors(err, goa.ValidatePattern("c", c2, "patternc"))
```